csrc/flash_attn/flash_api.cpp

```diff
@@ -261,6 +338,195 @@ void set_params_alibi(Flash_fwd_params &params, const c10::optional<at::Tensor>
 #endif
 }
 
+std::vector<at::Tensor>
+mha_fwd_sparse(at::Tensor &q,         // batch_size x seqlen_q x num_heads x head_size
```

Comment:
Would it be possible to add the varlen version of this kernel? I think that will help simplify the vLLM integration a lot.

Reply:
Sure, interface sparse_attn_varlen_func has been added.